framework/src/controller/ws/ws_server.ts

```diff
@@ -101,4 +134,30 @@ export class WSServer {
 			return null;
 		}, 3000);
 	}
+
+	private _handleSubscription(socket: WebSocketWithTracking, message: Partial<RequestObject>) {
+		const { params } = message;
+		if (!params || !Array.isArray(params.topics) || !params.topics.length) {
+			throw new Error('Invalid subscription message.');
+		}
+		if (!this._subscriptions[socket.url]) {
+			this._subscriptions[socket.url] = new Set<string>();
+		}
+		for (const eventName of params.topics) {
+			this._subscriptions[socket.url].add(eventName);
```

Comment:
In case if any client socket adds any number of eventNames as it's not checked, it can be used as a vulnerability to grow topics Set size. Maybe pass a list of registered events from Bus when passing message to `start` and use that to check before adding?